src/generators/clientFileGenerator.ts

```diff
@@ -65,23 +71,25 @@ export function generateClient(clientDetails: ClientDetails, project: Project) {
     })
   );
 
-  const requiredParams = clientDetails.parameters
-    .filter(param => param.isGlobal)
-    .filter(p => p.required);
+  const requiredParams = clientDetails.parameters.filter(
+    param => param.implementationLocation === ImplementationLocation.Client
+  );
 
   const clientConstructor = clientClass.addConstructor({
     docs: [
-      // TODO: Parameter list will need to be generated based on real
-      // client parameter list.
       `Initializes a new instance of the ${clientDetails.className} class.
 @param options The parameter options`
     ],
     parameters: [
-      ...requiredParams.map(p => ({ name: p.name, type: "any" })), // TODO Use the right type
+      ...requiredParams.map(p => ({
+        name: p.name,
+        hasQuestionToken: !p.required,
+        type: p.typeDetails.typeName
+      })),
       {
         name: "options",
         hasQuestionToken: true,
-        type: "any" // TODO Use the right type `Models.${clientDetails.className}Options`
+        type: `coreHttp.ServiceClientOptions`
```

Comment:
Should this be referring to the client-specific generated options type or is that not being generated yet?

Reply:
You're right we need to generate this type. Will send a followup PR for this